Gems/ROS2/Code/Include/ROS2/Manipulator/MotorizedJointComponent.h

```diff
@@ -0,0 +1,97 @@
+        void Deactivate() override;
+        static void Reflect(AZ::ReflectContext* context);
+
+        //! Set a setpoint (e.g. desired local position). The controller will follow it.
+        void SetSetpoint(float setpoint) override;
+
+        //! Get a setpoint
+        float GetSetpoint() override;
+
+        //! Get current control error. It is the difference between control value and measurement.
+        //! When the setpoint is reached this should be close to zero.
+        //! @returns control error, in meters for linear joints and in radians for angular joints.
+        float GetError() override;
+
+        //! Get current position from measurement.
+        //! @returns current position, in meters for linear joints and radians for angular joints.
+        float GetCurrentMeasurement() override;
+
+        //! Get a degree of freedom direction.
+        //! @returns direction of joint movement in global coordinates.
+        AZ::Vector3 GetDir() const
+        {
+            return m_jointDir;
+        };
+
+    private:
+        float ComputeMeasurement(AZ::ScriptTimePoint time);
+        void SetVelocity(float velocity, float deltaTime);
+        void ApplyLinVelAnimation(float velocity, float deltaTime);
+        void ApplyLinVelRigidBodyImpulse(float velocity, float deltaTime);
+        void ApplyLinVelRigidBody(float velocity, float deltaTime);
+        void OnTick(float deltaTime, AZ::ScriptTimePoint time) override;
+
+        AZ::Vector3 m_jointDir{ 0.f, 0.f, 1.f }; //!< Direction of joint movement in parent frame of reference, used to compute measurement.
+        AZ::Vector3 m_effortAxis{ 0.f, 0.f, 1.f }; //!< Direction of force or torque application in owning entity frame of reference.
+        AZStd::pair<float, float> m_limits{ -0.5f, 0.5f }; //!< limits of joint, the force is applied only when joint is within limits.
+        VehicleDynamics::PidConfiguration m_pidPos; //!< PID controller for position.
+
+        bool m_linear{ true }; //!< Linear mode. The force is applied through RigidBodyBus.
+        bool m_animationMode{ true }; //!< Use TransformBus (animation mode, no physics) instead of RigidBodyBus.
+
+        // TODO - remove test signal?
+        bool m_testSinusoidal{ true }; //!< Enable sinusoidal signal generator to setpoint (for tuning).
+        float m_sinAmplitude{ 0.5 }; //!< Amplitude of test signal generator.
+        float m_sinDC{ 0.25 }; //!< DC of test signal generator.
+        float m_sinFreq{ 0.1 }; //!< Frequency of test signal generator.
+
+        float m_zeroOffset{ 0.f }; //!< offset added to setpoint.
+        float m_setpoint{ 0 }; //!< Desired local position.
+        float m_error{ 0 }; //!< Current error (difference between control value and measurement).
+        float m_currentPosition{ 0 }; //!< Last measured position.
+        float m_currentVelocity{ 0 }; //!< Last measured velocity.
+        double m_lastMeasurementTime; //!< Last measurement time in seconds.
```

Comment:
This should be initialized to [zero initialized](https://en.cppreference.com/w/cpp/language/zero_initialization) as it can be checked via the path of `OnTick` -> `ComputedMeasurement` before it is initialized for the first time.
```suggestion
        double m_lastMeasurementTime{}; //!< Last measurement time in seconds.
```